src/components/FilterGroup.tsx

```diff
@@ -85,17 +96,62 @@ export function FilterGroup({
       {renderTitle()}
       <CollapsibleSection className={cssClasses.optionsContainer}>
         {searchable && <SearchInput className={cssClasses.searchInput} />}
-        {filterOptions.map(o => {
-          return (
-            <CheckboxOption
-              {...o}
-              key={o.displayName || o.value.toString()}
-              customCssClasses={cssClasses}
-            />
-          );
-        })}
+        <CheckboxOptions
+          filterOptions={filterOptions}
+          showMoreLimit={showMoreLimit}
+          cssClasses={cssClasses}
+          showAll={showAll} />
+        {isLimited &&
+          /* eslint-disable-next-line react-perf/jsx-no-new-function-as-prop */
+          <button className='text-primary py-1 text-sm' onClick={() => setShowAll(!showAll)}>
+            {showAll ? 'Show Less' : 'Show More'}
+          </button>
+        }
         {children}
       </CollapsibleSection>
     </FilterGroupProvider>
   );
 }
+
+function CheckboxOptions({
+  filterOptions,
+  showMoreLimit,
+  cssClasses,
+  showAll
+}: {
+  filterOptions: FilterOptionConfig[],
+  showMoreLimit: number,
+  cssClasses: CheckboxCssClasses,
+  showAll: boolean
+}) {
+  const searchUtilities = useSearchUtilities();
+  const { searchValue } = useFilterGroupContext();
+
+  const shouldRenderOption = (
+    option: { displayName?: string, value: FilterOptionConfig['value'] }
+  ) => {
+    if (isUndefined(option.displayName)) {
+      option.displayName = option.value.toString();
+    }
+
+    if (!searchUtilities.isCloseMatch(option.displayName, searchValue)) {
+      return false;
+    }
+
+    return true;
```

Comment:
I think we can directly return `searchUtilities.isCloseMatch(option.displayName, searchValue)` here?